File: products/signals/eval/self_driving/tasks/timezone-report-off-by-day/repo/acme_reports/analytics.py
import os
import json
import urllib.request


def capture(distinct_id: str, event: str, properties: dict | None = None) -> None:
    """Send a PostHog capture event. No-op unless POSTHOG_API_KEY is set."""
    api_key = os.environ.get("POSTHOG_API_KEY")
    if not api_key:
        return
    host = os.environ.get("POSTHOG_HOST", "https://us.i.posthog.com")
    payload = {
        "api_key": api_key,
        "event": event,
        "distinct_id": distinct_id,
        "properties": properties or {},
    }
    request = urllib.request.Request(
        f"{host}/i/v0/e/",
        data=json.dumps(payload).encode("utf-8"),
        headers={"Content-Type": "application/json"},
    )
    try:
        urllib.request.urlopen(request, timeout=2)
    except OSError:
        pass  # analytics must never break report generation

File: products/signals/eval/self_driving/tasks/checkout-coupon-case/repo/src/analytics.js
let client = null;

function getClient() {
  if (!client) {
    const { PostHog } = require("posthog-node");
    client = new PostHog(process.env.POSTHOG_API_KEY, {
      host: process.env.POSTHOG_HOST || "https://us.i.posthog.com",
    });
  }
  return client;
}

// Analytics is disabled (no-op) unless POSTHOG_API_KEY is configured.
function capture(distinctId, event, properties = {}) {
  if (!process.env.POSTHOG_API_KEY) return;
  getClient().capture({ distinctId, event, properties });
}

module.exports = { capture };
